src/test/java/DmainTests/ShoppingBasketTests.java

```diff
@@ -419,4 +431,169 @@ public void testGetShopId_whenBasketIsNotEmpty_shouldReturnShopId() {
+        date.setTime(0);
+        User buyer = new User("username1", "password1", "email1", date);
+        Shop shop = new Shop(1, "ownerUsername", "bank1", "address1");
+        ShoppingBasket shoppingBasket = new ShoppingBasket(shop);
+        Product product = new Product(1, "product1", Category.ELECTRONICS, 100.0);
+        product.updateProductQuantity(3);
+        shop.addProductToShop("ownerUsername", product);
+        Product product2 = new Product(2, "product2", Category.ELECTRONICS, 100.0);
+        product2.updateProductQuantity(1);
+        shop.addProductToShop("ownerUsername", product2);
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product2.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product2.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product2.getProductId());
+
+        // Act
+        boolean result = shoppingBasket.purchaseBasket();
+        
+        // Assert
+        assertFalse(result);
+        assertEquals(product.getProductQuantity(), 3);
+        assertEquals(product2.getProductQuantity(), 1);
+    }
+
+    @Test
+    public void testPurchaseBasket_whenBasketDoNotMeetsShopPolicyAndEverythingInStock_thenThrowsShopPolicyException() throws ProdcutPolicyException,
+     ShopPolicyException, ProductAlreadyExistsException, PermissionException, StockMarketException {
+        // Arrange
+        Date date = new Date();
+        date.setTime(0);
+        User buyer = new User("username1", "password1", "email1", date);
+        ShoppingBasket shoppingBasket = new ShoppingBasket(shopMock);
+        Product product = new Product(1, "product1", Category.ELECTRONICS, 100.0);
+        product.updateProductQuantity(3);
+        shopMock.addProductToShop("ownerUsername", product);
+        Product product2 = new Product(2, "product2", Category.ELECTRONICS, 100.0);
+        product2.updateProductQuantity(10);
+        shopMock.addProductToShop("ownerUsername", product2);
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product.getProductId());
+        shoppingBasket.addProductToShoppingBasket(buyer, product2.getProductId());
+        doThrow(new ShopPolicyException("Basket does not meet shop policy")).when(shopMock).ValidateBasketMeetsShopPolicy(shoppingBasket);
+
+        // Act & Assert
+        assertThrows(ShopPolicyException.class, () -> {
+            shoppingBasket.purchaseBasket();
+        });
+        assertEquals(product.getProductQuantity(), 3);
+        assertEquals(product2.getProductQuantity(), 10);
+    }
+
+    // @Test
```

Comment:
you can add @disable up of @test and them it will be ignored.
not need to do all in comment